tests/trace/test_evaluate.py

```diff
@@ -359,3 +361,67 @@ def score(target, output):
 
     result = asyncio.run(evaluation.evaluate(model))
     assert result == expected_eval_result
+
+
+def test_evaluate_table_lazy_iter(client):
+    """
+    The intention of this test is to show that an evaluation harness
+    lazily fetches rows from a table rather than eagerly fetching all
+    rows up front.
+    """
```

Comment:
I can see why you added this but it seems a bit overkill.  We know how the table mechanics work under the hood, and you showed it above by calling `len(ds)`

Reply:
the end of this test is actually quite interesting